### PR TITLE
homework01_sergio v2

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: homework01_sergio/vector.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "vector.h"
+
+void Vector::Add(const Vector& v)
+{
+    mX=mX + v.mX;
```
Comment (david-grs) [inline homework01_sergio/vector.cpp]: Style: space

Comment (david-grs) [inline homework01_sergio/vector.cpp]: We usually use `operator+=` instead of `x = x + `

File: homework01_sergio/vector.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    mY=mY + v.mY;
+}
+
+bool Vector::operator == (const Vector& v)
```
Comment (david-grs) [inline homework01_sergio/vector.cpp]: Style: `operator==`, not `operator == `

File: homework01_sergio/vector.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    return (mX == v.mX)
+        && (mY == v.mY);
+}
+
+Vector Vector::operator + (const Vector& v)
```
Comment (david-grs) [inline homework01_sergio/vector.cpp]: Same here, `operator+`

File: homework01_sergio/vector.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+{
+    return Vector(mX+v.mX, mY+v.mY);
```
Comment (david-grs) [inline homework01_sergio/vector.cpp]: and here, the opposite: leave some spaces between operands

File: homework01_sergio/vector.h [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#include <iostream>
```
Comment (david-grs) [inline homework01_sergio/vector.h:1]: Needed?

File: homework01_sergio/vector.cpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+Vector Vector::operator += (const Vector& v)
+{
+    this->Add(v);
```
Comment (david-grs) [inline homework01_sergio/vector.cpp]: `this->` isnt very popular in C++

File: homework01_sergio/vector.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    int GetY() const {return mY;}
+
+    void Add (const Vector&);
+
+    bool operator == (const Vector&);
+
+    Vector operator + (const Vector&);
+
+    Vector operator += (const Vector&);
```
Comment (david-grs) [inline homework01_sergio/vector.h]: Why returning by value?

File: homework01_sergio/vector.h [hunk 8]
```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+
+class Vector
+{
+public:
+    Vector(int x, int y)
+        : mX(x)
+        , mY(y)
```
Comment (david-grs) [inline homework01_sergio/vector.h]: Only one liners in header file

**Review by david-grs — APPROVED** [review]

File: homework01_sergio/vector.h (hunk 5 above)
Comment (ghost) [reply homework01_sergio/vector.h:1]: if removed it says
vector.h:19:17: error: ‘ostream’ in namespace ‘std’ does not name a type

**Pull request merged**